sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -314,11 +312,10 @@ public void setAttribute(String key, AttributeValue value) {
         logger.log(Level.FINE, "Calling setAttribute() on an ended Span.");
         return;
       }
-      if (value == null || (value.getType().equals(STRING) && value.getStringValue() == null)) {
-        if (attributes == null) {
-          return;
+      if (value == null || value.isEmpty()) {
```

Comment:
interesting. now that empty arrays are explicitly allowed, I wonder if "isEmpty" is no longer the right name. Maybe `isMissing` ?

Reply:
or maybe just `isNull()` since that's literally what we're checking for now.